test/dao.suite/patch-by-id.suite.js

```diff
@@ -0,0 +1,188 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+module.exports = function(dataSource, should, connectorCapabilities) {
+  describe('patchById', function() {
+    var personId, Person, StubUser;
+    before(function setupDatabase(done) {
+      Person = dataSource.define('Person', {
+        name: String,
+        age: { type: Number, index: true },
+      }, { forceId: true, strict: true });
+      // A simplified implementation of LoopBack's User model
+      StubUser = dataSource.createModel('StubUser', { password: String }, { forceId: true });
+      StubUser.setter.password = function(plain) {
+        var hashed = false;
+        if (!plain) return;
+        var pos = plain.indexOf('-');
+        if (pos !== -1) {
+          var head = plain.substr(0, pos);
+          var tail = plain.substr(pos + 1, plain.length);
+          hashed = head.toUpperCase() === tail;
+        }
+        if (hashed) return;
+        this.$password = plain + '-' + plain.toUpperCase();
```

Comment:
Can we please move the body of this setter function to a shared method/function? Or perhaps extract the whole `StubUser`